File: 2015/Python/01/part_one.py
"""
Advent of Code 2015 day 01 part one
"""

import sys

INPUT_FILE = "../inputs/01-input.txt"

if __name__ == '__main__':
    floor = 0
    index = 0
    directions = {'(': 1, ')': -1}

    try:
        with open(INPUT_FILE, 'r', encoding='UTF-8') as file:
            while True:
                inst = file.read(1)
                if not inst:
                    break # EOF
                if inst[0] in directions:
                    floor += directions[inst[0]]
                else:
                    sys.stderr.write(f"Invalid character found at position '{index+1}'\n")
                    sys.exit(1)
                index += 1
    except IOError as e:
        sys.stderr.write(f"{e}\n")
        sys.exit(1)

    print(f"The right floor is '{floor}'")

File: 2015/Lua/01/part_one.lua
-- Advent of Code 2015 day 01 part one

io.input("../inputs/01-input.txt")

local floor = 0
local index = 1
local directions = { ['('] = 1, [')'] = -1 }

while true do
    local inst = io.read(1)
    if (inst == nil) then
        break
    elseif directions[inst] ~= nil then
        floor = floor + directions[inst]
    else
        io.stderr:write("Invalid character found at position '" .. index .. "'\n")
        os.exit(1)
    end
    index = index + 1
end

print("The right floor is '" .. floor .. "'")
